frontend/src/components/accountdetails/AccountDetailsPage.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import {Link} from 'react-router-dom'
+import AccountDetailsTable from './AccountDetailsTable'
+
+export default class AccountDetailsPage extends React.Component {
+  constructor(props){
+    super(props)
+
+    this.state = {}
+
+    this.onDeleteAccount = this.onDeleteAccount.bind(this)
+    this.onDeposit = this.onDeposit.bind(this)
+    this.onWithdraw = this.onWithdraw.bind(this)
+    this.onAccountHistory = this.onAccountHistory.bind(this)
+    this.accountFilter = this.accountFilter.bind(this)
+  }
+
+  componentDidMount() {
+    this.setState({loading: true})
+    this.props.loadAccountDetails(this.props.match.params.id).then(response => {
+      this.setState({loading: false})
+    }).catch((error) => {
+      this.setState({loadng: false})
+    })
+  }
+
+  render() {
+    if (!this.props.accountDetails) {
+      return <p>Bad Account ID. Please go back.</p> 
+    }
+    const { loading } = this.state
```

Comment:
The loading check should be performed first, as accountDetails is undefined when loading starts and loading message will not be shown.